crates/forge-runner/src/test_case_summary.rs

```diff
@@ -195,13 +195,26 @@ impl TestCaseSummary<Single> {
         let msg = extract_result_data(&run_result, &test_case.expected_result);
         match run_result.value {
             RunResultValue::Success(_) => match &test_case.expected_result {
-                ExpectedTestResult::Success => TestCaseSummary::Passed {
-                    name,
-                    msg,
-                    arguments,
-                    test_statistics: (),
-                    gas_info: gas,
-                },
+                ExpectedTestResult::Success => {
+                    if let Some(available_gas) = test_case.available_gas {
+                        if gas > available_gas as u128 {
+                            return TestCaseSummary::Failed {
+                                name,
+                                msg: Some("\n\tavailable_gas exceeded".to_string()),
```

Comment:
I don't think that this constructor is a right place to put this logic

Reply:
It should be placed in own function and ran elsewhere, after `run_test_case` result is retrieved